### ISSUE TITLE
Encoder: ML-driven cost model / config selection

### ISSUE THREAD

**Issue by lilith** [imazen/zenwebp#40]
**Future work** — discussed during the libwebp-parity sprint (audit branch `audit/libwebp-parity-2026-04-26`).

## What

Train and deploy a small ML model that, given an input image (or a cheap feature digest of it), predicts the best `CostModel` and per-image tuning parameters (sns, filter, segments, partition_limit, q-shift) for a chosen objective.

## Why

Brute-force per-image config search (separate issue) is high-quality but slow. An ML model trained on the search results can make the same prediction in microseconds, dropping the per-image cost from O(N×encode) to O(1×encode + cheap-inference).

This is what `zentract` was built for — runtime ONNX inference plugin system for image-processing decisions.

## Sketch

1. **Feature extraction** — cheap per-image features computable in < 1ms:
   - Dimensions, aspect ratio
   - Color count / palette histogram entropy
   - DCT energy distribution (low-freq vs. high-freq)
   - Edge density (Sobel magnitude histogram)
   - Saliency map (zensally)
2. **Training corpus** — generate `(features, best_config)` pairs by running the brute-force config search across CID22, CLIC2025, imagemagick, screenshot collections, etc.
3. **Model** — small gradient-boosted decision tree (xgboost-class, ~100 trees, KB-scale ONNX) or a tiny transformer. Output: predicted (preset, sns, filter, segments, cost_model, q_offset).
4. **Inference** — via `zentract` plugin loaded at encoder startup. Optional, defaulting to current heuristic Auto preset.
5. **Validation** — A/B against brute-force search on held-out images. Target: within 1% size of brute-force at 100× the speed.

## API shape

```rust
EncoderConfig::with_ml_predictor("zentract://webp-config-v1.onnx")
    .target_quality(SsimQ::Excellent)
```

## Dependencies

- Brute-force config search (separate issue) — provides training data
- `zentract` plugin infra — already in workspace (`~/work/zen/zentract/`)
- `zensally` for saliency features — already exists
- A clean cost-model API surface (`CostModel` enum from #33) — in flight

## Status

Backlog. No implementation in this session.
